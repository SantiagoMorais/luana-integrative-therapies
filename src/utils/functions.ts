import { IDefaultTheme, ISectionsTheme, sectionsTheme, theme } from "@styles/theme";
import { Location } from "react-router-dom";

export const handlePageTheme = (
   location: string
): ISectionsTheme | IDefaultTheme => {
   const sectionTheme = sectionsTheme.find(
      (theme) => theme.id === location
   );

   if (location === "equilibrium" || location === "segredos-da-lua") {
      return sectionTheme ?? theme;
   } else return theme;
};

export const locationName = (location: Location) => {
    return location.pathname.slice(1).split("/")[0];
}
