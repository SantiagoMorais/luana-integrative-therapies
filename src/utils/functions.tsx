import { BrowserRouter, Location } from "react-router-dom";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@contexts/themeContext";

export const locationName = (location: Location) => {
   return location.pathname.slice(1).split("/")[0];
};

export const renderWithProviders = (ui: React.ReactElement) => {
   return render(
      <BrowserRouter>
         <ThemeProvider>{ui}</ThemeProvider>
      </BrowserRouter>
   );
};
