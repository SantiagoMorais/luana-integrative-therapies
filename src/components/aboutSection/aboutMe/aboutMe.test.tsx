import { screen } from "@testing-library/react";
import { AboutMe } from ".";
import { renderWithProviders } from "@utils/functions";

describe("<AboutMe />", () => {
   it("should render the component correctly", () => {
      renderWithProviders(<AboutMe />);
      const title = screen.getByRole("heading", { name: /sobre mim/i });
      expect(title).toBeInTheDocument();
   });

   it("should render the image correctly", () => {
      renderWithProviders(<AboutMe />);
      const personalImage = screen.getAllByAltText("Foto luana no consultÃ³rio");
      personalImage.map((img) => expect(img).toBeInTheDocument());
      expect(personalImage).toHaveLength(3);
   });

   it("should render the text correctly", () => {
      renderWithProviders(<AboutMe />);
      const text = document.querySelectorAll(".paragraph");
      expect(text).toMatchSnapshot();
   });
});
