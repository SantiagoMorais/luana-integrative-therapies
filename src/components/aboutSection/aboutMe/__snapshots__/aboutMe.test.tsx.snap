// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AboutMe /> > should render the text correctly 1`] = `
NodeList [
  <span
    class="paragraph"
  >
    Muito prazer, meu nome é Luana, sou cirurgiã-dentista e terapeuta integrativa. Formei-me em 2018 pela UFMG em odontologia e comecei a atuar imediatamente. A partir daí começou minha busca incessante por algo que nem eu mesma sabia ao certo o que era. Sempre fui apaixonada por reabilitar sorrisos, mas sentia falta de algo em meus atendimentos. Apesar de sempre optar por um atendimento humanizado, com tecnologias modernas, havia algo que ainda não estava conseguindo atingir em meu paciente: a vontade de sorrir.
  </span>,
  <span
    class="paragraph"
  >
    Em 2022, descobri o universo das terapias integrativas como paciente e fui em busca de entender melhor sobre a medicina tradicional chinesa. Desde então, não passo um dia sequer sem estudar um pouco mais sobre as terapias integrativas.
  </span>,
  <span
    class="paragraph"
  >
    Busquei me especializar nessa área, pois, assim como senti os benefícios e efeitos das práticas integrativas, queria proporcionar isso também para meus pacientes. Realizei pós-graduações em várias áreas dentro das práticas integrativas, com foco na área terapêutica e estética, e venho me aperfeiçoando a cada dia. Através de muito estudo teórico e prático, desenvolvi um método de atendimento no qual realizo diversas práticas integrativas na mesma sessão em prol do mesmo objetivo, e comecei a reparar na eficácia e durabilidade dos resultados.
  </span>,
  <span
    class="paragraph"
  >
    O segredo da nossa saúde, beleza e bem-estar está no equilíbrio físico, emocional e energético. Então, além de atuar na estética do sorriso, ajudo meu paciente a resgatar a vontade de sorrir! Hoje me sinto mais completa como profissional e muito mais realizada na minha vida pessoal!
  </span>,
]
`;
